File: java/ConsultarIe.java
// Consulta a Inscrição Estadual (IE) de um CNPJ pela API da CNPJAPI (https://cnpjapi.com.br).
// Recurso premium (plano com IE). Java 17+ (o HttpClient e nativo desde o 11). Arquivo unico:
// Uso: CNPJAPI_KEY=cnpj_sua_chave java ConsultarIe.java 00776574000156 SP
//      (a UF e opcional; sem ela, faz a varredura nacional)
//
// A resposta e JSON com { cnpj, fonte, cache_hit, as_of, resultados:[...] };
// aqui imprimimos o corpo cru. Para mapear em objetos, use uma lib JSON (Jackson, Gson).

import java.net.URI;
import java.net.URLEncoder;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.nio.charset.StandardCharsets;

public class ConsultarIe {
    public static void main(String[] args) throws Exception {
        String cnpj = args.length > 0 ? args[0] : "00776574000156";
        String uf = args.length > 1 ? args[1] : "";
        String apiKey = System.getenv().getOrDefault("CNPJAPI_KEY", "cnpj_sua_chave");

        String url = "https://api.cnpjapi.com.br/consulta/ie/" + cnpj;
        if (!uf.isEmpty()) {
            url += "?uf=" + URLEncoder.encode(uf, StandardCharsets.UTF_8);
        }

        HttpClient client = HttpClient.newHttpClient();
        HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(url))
                .header("Authorization", "Bearer " + apiKey)
                .GET()
                .build();

        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());

        if (response.statusCode() != 200) {
            System.err.println("Falha na consulta de IE: HTTP " + response.statusCode());
            System.exit(1);
        }

        System.out.println(response.body());
    }
}

File: php/consultar-ie.php
<?php

// Consulta a Inscrição Estadual (IE) de um CNPJ pela API da CNPJAPI (https://cnpjapi.com.br).
// Recurso premium (plano com IE). Requer PHP 8.0+ com a extensao cURL.
// Uso: CNPJAPI_KEY=cnpj_sua_chave php consultar-ie.php 00776574000156 SP
//      (a UF e opcional; sem ela, faz a varredura nacional)

$cnpj = $argv[1] ?? '00776574000156';
$uf = $argv[2] ?? '';
$apiKey = getenv('CNPJAPI_KEY') ?: 'cnpj_sua_chave';

$url = "https://api.cnpjapi.com.br/consulta/ie/{$cnpj}";
if ($uf !== '') {
    $url .= '?uf=' . urlencode($uf);
}

$ch = curl_init($url);
curl_setopt_array($ch, [
    CURLOPT_RETURNTRANSFER => true,
    CURLOPT_HTTPHEADER => ["Authorization: Bearer {$apiKey}"],
    CURLOPT_TIMEOUT => 15,
]);

$body = curl_exec($ch);
$status = curl_getinfo($ch, CURLINFO_HTTP_CODE);
curl_close($ch);

if ($status !== 200) {
    fwrite(STDERR, "Falha na consulta de IE: HTTP {$status}\n");
    exit(1);
}

$dados = json_decode($body, true);
if (empty($dados['resultados'])) {
    echo 'Nenhuma inscricao estadual encontrada.', PHP_EOL;
} else {
    foreach ($dados['resultados'] as $ie) {
        $numero = $ie['ie'] !== '' ? $ie['ie'] : '(nao-contribuinte)';
        echo "{$ie['uf']}: {$numero} - {$ie['situacao']}", PHP_EOL;
    }
}
